src/engine/engine_wi.c

```diff
@@ -18,11 +18,49 @@
 
 static int ocf_write_wi_update_and_flush_metadata(struct ocf_request *req);
 
-static const struct ocf_io_if _io_if_wi_flush_metadata = {
+static const struct ocf_io_if _io_if_wi_update_metadata = {
 		.read = ocf_write_wi_update_and_flush_metadata,
 		.write = ocf_write_wi_update_and_flush_metadata,
 };
 
+int _ocf_write_wi_next_pass(struct ocf_request *req)
+{
+	ocf_req_unlock_wr(req);
+
+	if (ocf_engine_is_hit(req)) {
+		/* second pass not needed for hit -
+			mark it as completed */
+		req->wi_second_pass = true;
+	}
+
+	if (req->wi_second_pass) {
+		req->complete(req, req->error);
+		ocf_req_put(req);
+
+		return 0;
+	}
```

Comment:
This is not needed, as we already checked for miss before entering this function.

Reply:
This check is needed when this function called from IO queue after metadata flush

Reply:
hit check is indeed seems to be redundant